core/src/main/java/com/linecorp/armeria/common/Cookies.java

```diff
@@ -13,32 +13,57 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-package com.linecorp.armeria.server.annotation;
+package com.linecorp.armeria.common;
 
 import static java.util.Objects.requireNonNull;
 
 import java.util.Set;
 
 import com.google.common.collect.ImmutableSet;
 
-import io.netty.handler.codec.http.cookie.Cookie;
-
 /**
- * An interface which holds decoded {@link Cookie} instances for an HTTP request.
+ * An immutable {@link Set} of {@link Cookie}s.
  */
 public interface Cookies extends Set<Cookie> {
 
+    /**
+     * Returns an empty {@link Set} if {@link Cookie}s.
+     */
+    static Cookies empty() {
+        return DefaultCookies.EMPTY;
+    }
+
     /**
      * Creates an instance with a copy of the specified set of {@link Cookie}s.
      */
     static Cookies of(Cookie... cookies) {
-        return new DefaultCookies(ImmutableSet.copyOf(requireNonNull(cookies, "cookies")));
+        requireNonNull(cookies, "cookies");
+        if (cookies.length == 0) {
```

Comment:
This is fine too but we could do `return of(ImmutableSet.copyOf(cookies)` for less code, and no change in allocations since `copyOf` of empty array returns empty set, and `copyOf` of `ImmutableSet` returns the same instance.

Reply:
Sure. No problem.